source/es5-wrappers.ts

```diff
@@ -0,0 +1,124 @@
+	blockHistory: BlockHistory | null,
+	newBlock: Block,
+	onBlockAdded: (block: Block, callback: (error?: Error) => void) => void,
+	onBlockRemoved: (block: Block, callback: (error?: Error) => void) => void,
+	blockRetention: number,
+	callback: (error?: Error, newHistory?: BlockHistory) => void,
+): void {
+	const wrappedGetBlockByHash = (hash: string): Promise<Block | null> => new Promise<Block | null>((resolve, reject) => {
+		getBlockByHash(hash, (error, block) => {
+			if (error) reject(error);
+			else resolve(block);
+		});
+	});
+	const wrappedOnBlockAdded = (block: Block): Promise<void> => new Promise<void>((resolve, reject) => {
+		onBlockAdded(block, (error) => {
+			if (error) reject(error);
+			else resolve();
+		});
+	});
+	const wrappedOnBlockRemoved = (block: Block): Promise<void> => new Promise<void>((resolve, reject) => {
+		onBlockRemoved(block, (error) => {
+			if (error) reject(error);
+			else resolve();
+		});
+	});
+	reconcileBlockHistory(wrappedGetBlockByHash, blockHistory, newBlock, wrappedOnBlockAdded, wrappedOnBlockRemoved, blockRetention)
+		.then(newBlockHistory => callback(undefined, newBlockHistory))
+		.catch(error => callback(error, undefined));
+}
+
+export function reconcileLogHistoryWithAddedBlockEs5(
+	getLogs: (filterOptions: FilterOptions[], callback: (error?: Error, logs?: Log[]) => void) => void,
+	logHistory: LogHistory | null,
+	newBlock: Block,
+	onLogAdded: (log: Log, callback: (error?: Error) => void) => void,
+	filters: Filter[],
+	blockRetention: number,
+	callback: (error?: Error, newHistory?: LogHistory) => void,
+): void {
+	const wrappedGetLogs = (filterOptions: FilterOptions[]): Promise<Log[]> => new Promise<Log[]>((resolve, reject) => {
+		getLogs(filterOptions, (error, logs) => {
+			if (error) reject(error);
+			else resolve(logs);
+		})
+	});
+	const wrappedOnLogAdded = (log: Log): Promise<void> => new Promise<void>((resolve, reject) => {
+		onLogAdded(log, error => {
+			if (error) reject(error);
+			else resolve();
+		});
+	});
+	reconcileLogHistoryWithAddedBlock(wrappedGetLogs, logHistory, newBlock, wrappedOnLogAdded, filters, blockRetention)
+		.then(newLogHistory => callback(undefined, newLogHistory))
+		.catch(error => callback(error, undefined));
+}
+
+export function reconcileLogHistoryWithRemovedBlockEs5(
+	logHistory: LogHistory,
+	removedBlock: Block,
+	onLogRemoved: (log: Log, callback: (error?: Error) => void) => void,
+	callback: (error?: Error, logHistory?: LogHistory) => void,
+): void {
+	const wrappedOnLogRemoved = (log: Log): Promise<void> => new Promise<void>((resolve, reject) => {
+		onLogRemoved(log, error => {
+			if (error) reject(error);
+			else resolve();
+		});
+	});
+	reconcileLogHistoryWithRemovedBlock(logHistory, removedBlock, wrappedOnLogRemoved)
+		.then(logHistory => callback(undefined, logHistory))
+		.catch(error => callback(error, undefined));
+}
+
+export function reconcileBlocksAndLogsEs5(
```

Comment:
I have renamed these to `reconcileBlockHistoryWithCallback` and similar.  The promise methods are the real methods, so I would prefer to not change their name away from the base.  These methods are just shim wrappers to allow for legacy style usage, but the hope would be that eventually these methods will be removed once we decide to end-of-life callback style support.